examples/data-objects/shared-text/src/component.ts

```diff
@@ -134,25 +135,25 @@ export class SharedTextRunner
                     newString.insertMarker(newString.getLength(), marker.refType, marker.properties);
                 }
             }
-            this.rootView.set("text", newString.handle);
+            this.rootView.set(textSharedStringId, newString.handle);
 
             insights.set(newString.id, this.sharedTextDocument.createMap().handle);
 
             // The flowContainerMap MUST be set last
 
             const flowContainerMap = this.sharedTextDocument.createMap();
-            this.rootView.set("flowContainerMap", flowContainerMap.handle);
+            this.rootView.set(flowContainerMapId, flowContainerMap.handle);
 
             insights.set(newString.id, this.sharedTextDocument.createMap().handle);
         }
 
         debug(`collabDoc loaded ${this.runtime.id} - ${performance.now()}`);
         debug(`Getting root ${this.runtime.id} - ${performance.now()}`);
 
-        await this.rootView.wait("flowContainerMap");
+        await mapWait(this.rootView, flowContainerMapId);
```

Comment:
Is this wait really required? Glancing from the code, the branch for !existing above suggest that this code is trying to deal with creation flow, and creation flow should leverage "attach final state" concept, not "slowly initial your state while it's visible in a file", as later guarantees things will deadlock forever if that dies.

Reply:
If I got it wrong, I'd rather fix samples, or remove them (same for others). I'd guess (without deeply looking at each example) is that each wait is likely either not needed (can be safely removed) or actually represents a race/bug that would be nice to address vs. leaving it around.

Reply:
My priority with this change was prepping for removal of the `.wait()` method from map/directory -- the sooner we do that the fewer new usages can spring up.

As a result, I didn't exhaustively run down all usages to determine if a switch to `.get()` was safe in cases where it became non-obvious.  Here I observed that we float the `SharedTextRunner.load()` promise below, in `instantiateDataStore()`, and so I decided to err on the side of caution that the floated promise might result in inconsistent behavior.  But it's totally possible with further investigation that it's not really needed.

Reply:
I played around with this a bit more (did a little cleanup of shared-text while I was there).

This wait is beneficial for the reason I mention above, that we float the `SharedTextRunner.load()` promise.  This float only works today because of two things:

1. The `flowContainerMap` is set last (as noted in the comment above) serving as a makeshift signal to other clients that initialization is actually done.  The `.wait()` then ensures those other clients don't try to access anything before that signal is received.
2. `webpack-fluid-loader` currently awaits a request for the component before attaching the container, effectively un-floating the promise before container attach.  Not all hosts will have this guarantee.

So if the `.wait()` is removed here, it will continue to work fine for `webpack-fluid-loader` but would turn the example into a non-robust initialization flow that probably fails for other hosts.

I'd suggest we open an issue to fix this and the other examples with similar flows (maybe convert into Aqueduct examples), I don't want to creep the scope of this change further.